torch/utils/ordered_set.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+from collections.abc import MutableSet, Set as AbstractSet
+from typing import Any, cast, Iterable, Iterator, List, Optional, Tuple, Type, TypeVar
+
+T = TypeVar("T")
+T_co = TypeVar("T_co", covariant=True)
+
+__all__ = ["OrderedSet"]
+
+
+class OrderedSet(MutableSet[T]):
+    """
+    Insertion ordered set, similar to OrderedDict.
+    """
+
+    __slots__ = ("_dict",)
+
+    def __init__(self, iterable: Optional[Iterable[T]] = None):
+        self._dict: dict[T, None] = {}
+        if iterable is not None:
+            self._dict = {k: None for k in iterable}
+
+    #
+    # Required overriden abstract methods
+    #
+
+    def __contains__(self, elem: object) -> bool:
+        return elem in self._dict
+
+    def __iter__(self) -> Iterator[T]:
+        yield from self._dict.keys()
+
+    def __len__(self) -> int:
+        return len(self._dict)
+
+    def add(self, elem: T) -> None:
+        self._dict[elem] = None
+
+    def discard(self, elem: T) -> None:
+        self._dict.pop(elem, None)
+
+    def clear(self) -> None:
+        # overridden because MutableSet impl is slow
+        self._dict.clear()
+
+    # Unimplemented set() methods in _collections_abc.MutableSet
```

Comment:
Implement `OrderedSet.move_to_end()` from `OrderedDict.move_to_end()`

```suggestion
    # Unimplemented set() methods in _collections_abc.MutableSet
    
    def move_to_end(self, value: T, last: bool = True) -> None:
        self._dict.move_to_end(value, last=last)
```